controllers/controller/workspacerouting/workspacerouting_controller.go

```diff
@@ -85,15 +86,11 @@ func (r *WorkspaceRoutingReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 	}
 
 	solver, err := getSolverForRoutingClass(instance.Spec.RoutingClass)
-	if err != nil {
-		// No solver in this controller for this routingClass
-		reqLogger.Info("Could not find solver for routing class; assuming external routing", "routingClass", instance.Spec.RoutingClass)
-		instance.Annotations = maputils.Append(instance.Annotations, config.WorkspaceExternalRoutingAnnotation, "true")
-		updateErr := r.Update(ctx, instance)
-		if updateErr != nil && errors.IsConflict(err) {
-			return reconcile.Result{Requeue: true}, nil
-		}
-		return reconcile.Result{}, updateErr
+	if err != nil && !errors.Is(err, externalRoutingError) {
```

Comment:
The `externalRoutingError` should never happen right ?

Reply:
Removed

Reply:
Ha, turns out it *can* happen, as we still trigger events for workspace services regardless of routingClass